File: src/platform/tauri/exportJobs.ts
import { invokeNativeCommand } from './nativeCommands';
import type { PrismCommandError } from './result';

export type ExportJobStatus = 'running' | 'completed' | 'failed' | 'cancelled' | string;

export interface ExportJobDto {
  id: string;
  format: string;
  documentPath?: string | null;
  outputPath?: string | null;
  status: ExportJobStatus;
  stage: string;
  message: string;
  createdAt: number;
  updatedAt: number;
  error?: PrismCommandError | null;
  cancelRequested: boolean;
}

export interface CreateExportJobInputDto {
  format: string;
  documentPath?: string | null;
  outputPath?: string | null;
  stage?: string | null;
  message?: string | null;
}

export interface UpdateExportJobInputDto {
  id: string;
  outputPath?: string | null;
  stage?: string | null;
  message?: string | null;
}

export interface CompleteExportJobInputDto {
  id: string;
  outputPath?: string | null;
  message?: string | null;
}

export interface FailExportJobInputDto {
  id: string;
  stage?: string | null;
  message?: string | null;
  error: PrismCommandError;
}

export function createExportJobNative(input: CreateExportJobInputDto) {
  return invokeNativeCommand<unknown>('create_export_job', { input });
}

export function updateExportJobNative(input: UpdateExportJobInputDto) {
  return invokeNativeCommand<unknown>('update_export_job', { input });
}

export function completeExportJobNative(input: CompleteExportJobInputDto) {
  return invokeNativeCommand<unknown>('complete_export_job', { input });
}

export function failExportJobNative(input: FailExportJobInputDto) {
  return invokeNativeCommand<unknown>('fail_export_job', { input });
}

export function cancelExportJobNative(jobId: string) {
  return invokeNativeCommand<unknown>('cancel_export_job', { jobId });
}

export function getExportJobNative(jobId: string) {
  return invokeNativeCommand<unknown>('get_export_job', { jobId });
}

export function listExportJobsNative() {
  return invokeNativeCommand<unknown>('list_export_jobs');
}

File: src-tauri/src/commands/export_jobs.rs
use tauri::State;

use crate::domain::error::PrismResult;
use crate::domain::export_job::{
    self, CompleteExportJobInput, CreateExportJobInput, ExportJobDto, ExportJobStore,
    FailExportJobInput, UpdateExportJobInput,
};

#[tauri::command]
pub fn create_export_job(
    store: State<'_, ExportJobStore>,
    input: CreateExportJobInput,
) -> PrismResult<ExportJobDto> {
    export_job::create_export_job(&store, input)
}

#[tauri::command]
pub fn update_export_job(
    store: State<'_, ExportJobStore>,
    input: UpdateExportJobInput,
) -> PrismResult<ExportJobDto> {
    export_job::update_export_job(&store, input)
}

#[tauri::command]
pub fn complete_export_job(
    store: State<'_, ExportJobStore>,
    input: CompleteExportJobInput,
) -> PrismResult<ExportJobDto> {
    export_job::complete_export_job(&store, input)
}

#[tauri::command]
pub fn fail_export_job(
    store: State<'_, ExportJobStore>,
    input: FailExportJobInput,
) -> PrismResult<ExportJobDto> {
    export_job::fail_export_job(&store, input)
}

#[tauri::command]
pub fn cancel_export_job(
    store: State<'_, ExportJobStore>,
    job_id: String,
) -> PrismResult<ExportJobDto> {
    export_job::cancel_export_job(&store, job_id)
}

#[tauri::command]
pub fn get_export_job(
    store: State<'_, ExportJobStore>,
    job_id: String,
) -> PrismResult<ExportJobDto> {
    export_job::get_export_job(&store, job_id)
}

#[tauri::command]
pub fn list_export_jobs(store: State<'_, ExportJobStore>) -> PrismResult<Vec<ExportJobDto>> {
    export_job::list_export_jobs(&store)
}
